dvc/remote/base.py

```diff
@@ -625,15 +625,10 @@ def cache_exists(self, checksums, jobs=None):
         """
         progress_callback = ProgressCallback(len(checksums))
 
-        def exists_with_progress(chunks):
-            return self.batch_exists(chunks, callback=progress_callback)
-
-        if self.no_traverse and hasattr(self, "batch_exists"):
+        if self.no_traverse:
             with ThreadPoolExecutor(max_workers=jobs or self.JOBS) as executor:
                 path_infos = [self.checksum_to_path_info(x) for x in checksums]
-                chunks = to_chunks(path_infos, num_chunks=self.JOBS)
-                results = executor.map(exists_with_progress, chunks)
-                in_remote = itertools.chain.from_iterable(results)
+                in_remote = list(executor.map(self.exists, path_infos))
                 ret = list(itertools.compress(checksums, in_remote))
```

Comment:
`self.exists` returns true or false, but `batch_exists` returned paths, so we need to adjust the logic.

Reply:
Ah, I'm wrong, map preserves the order anyway, so this should work as intended.

Reply:
Actually, I believe that logic started to work properly with this change.  `in_remote` behaves like bitmask, and `compress` can work with boolean array to filter the list. Earlier `in_remote` was list of checksums, and compress interprets [selector as boolean values](https://docs.python.org/3.1/library/itertools.html#itertools.compress).